### ISSUE TITLE
assuming image dimensions in ctlra_cairo.c

### ISSUE THREAD

**Issue by Be-ing** [openAVproductions/openAV-Ctlra#104]
https://github.com/openAVproductions/openAV-Ctlra/blob/881798bbadd04890c378f6e6079a8710d1fa3566/ctlra/ctlra_cairo.c#L61

Shouldn't this be using the width and height parameters of the function?

**Comment by harryhaaren** [comment]
Yes. Screen support is WIP, and there are lots of assumptions in the code for NI hardware with specific pixel dimensions right now.

[Edit]
To be clearer - each device will require its own "to device format" pixel conversion function. Here there are a few options to play with Cairo contexts at various different RGB888, or RGBA8888 formats, and convert the results into the device-speicfic formats. This needs to be abstracted via a function-pointer to enable all devices. At that point it must also scale to other pixel resolutions etc.
